stream_registry.go

```diff
@@ -28,6 +29,7 @@ type provider interface {
 	streamDefinition() *StreamDefinition
 	sendLoop(strm grpc.ServerStream, peer Peer)
 	streamType() stream.StreamType
+	sendHelloMessage(strm grpc.ServerStream, peer Peer) error
```

Comment:
does it make sense to export this one? I see this more like an internal hidden implementation detail

Reply:
provider is an internal interface
it is used to have a common interface between the simple streams and the getAndWatch streams